### PR TITLE
DX-434: allow for creating custom Packaging objects on the fly

### PR DESCRIPTION
**Pull request by rkrauskopf**
Docs change:
https://github.com/ShipEngine/connect-website/pull/47

Jira ticket:
https://auctane.atlassian.net/browse/DX-434

### CODE REVIEW

File: src/internal/carriers/packaging.ts [hunk 1]
```diff
@@ -26,21 +26,23 @@ export class Packaging extends DefinitionIdentifier implements IPackaging {
...
+      // Make this object immutable
+      this[_private] = {
+        app
+      };
 
-    // Make this object immutable
-    hideAndFreeze(this);
-
-    app[_internal].references.add(this);
+      hideAndFreeze(this);
```
Comment (JamesMessinger) [inline src/internal/carriers/packaging.ts:43]: `hideAndFreeze` needs to be called, even if the `app` is undefined

File: src/internal/carriers/pickups/pickup-package.ts [hunk 2]
```diff
@@ -38,8 +39,15 @@ export class PickupPackage extends PackageIdentifierBase implements IPickupPacka
...
+        pkg = new Packaging(
+          {
+            id: v4(),
+            name: pojo.packaging,
+            description: pojo.packaging,
+            code: "custom"
+          }
+        );
       }
     }
```
Comment (JamesMessinger) [inline src/internal/carriers/pickups/pickup-package.ts:52]: This same block of code (the entire try...catch block) is repeated in 4 places.  Move it to a utility function instead.

**Review by JamesMessinger — CHANGES_REQUESTED** [review]

**Comment by rkrauskopf** [comment]
@JamesMessinger another random thought on this, what if the user has a `Packaging` definition already defined whose code is set to `custom`? I would think we would want some validation in the sdk around that scenario

File: src/internal/carriers/pickups/pickup-package.ts (hunk 2 above)
Comment (rkrauskopf) [reply src/internal/carriers/pickups/pickup-package.ts]: done.

**Pull request merged**